src/repl.jl

```diff
@@ -58,10 +58,10 @@ handle("resetprompt") do linebreak
   nothing
 end
 
-current_prompt = juliaprompt
+const current_prompt = Ref{String}(juliaprompt)
 
 function hideprompt(f)
-  isREPL() || return f()
+  (isREPL() && Base.active_repl.mistate isa REPL.LineEdit.MIState) || return f()
```

Comment:
this change is because I encountered an error that says `mistate` is still `nothing` if we start Julia via command like `julia-client: run-all`.

I suspect this happens `hideprompt` can be called before `active_repl.interface` is fully setup:
- I still not tried to fix our script loading/activation order, but I think adding the snippet at our `atreplinit` may solve this all ?
```jl
if !isdefined(repl, :interface)
    repl.interface = REPL.setup_interface(repl)
end
```

Reply:
Just put that check into `isREPL` -- we're assuming that that's true all over the place anyways.

Reply:
I didn't do that since then [this part](https://github.com/JunoLab/Atom.jl/blob/master/src/Atom.jl#L17-L33) wouldn't get fired since it's supposed to be called before `run_main_repl` is called: https://github.com/JuliaLang/julia/blob/master/base/client.jl#L312

I special cased it in the last commit, and does it sound fine ?

Reply:
`run_main_repl` sets up `active_repl.mistate` but [`connected`](https://github.com/JunoLab/Atom.jl/blob/master/src/Atom.jl#L16) can be called before `run_main_repl`, can't it ?